src/databricks/labs/ucx/installer/logs.py

```diff
@@ -0,0 +1,157 @@
+    while len(line) > 0 and match is None:
+        multi_line_message += "\n" + line.rstrip()
+        line = log.readline()
+        match = pattern.match(line)
+    return line, match, multi_line_message
+
+
+def parse_logs(log: TextIO) -> Iterator[PartialLogRecord]:
+    """Parse the logs to retrieve values for PartialLogRecord fields.
+
+    Args:
+         log (TextIO): The log file IO.
+    """
+    # This regex matches the log format defined in databricks.labs.ucx.installer.logs.TaskLogger
+    log_format = r"(\d+):(\d+):(\d+)\s(\w+)\s\[(.+)\]\s\{\w+\}\s(.+)"
+    pattern = re.compile(log_format)
+
+    line = log.readline()
+    match = pattern.match(line)
+    while len(line) > 0:
+        assert match is not None
+        *groups, message = match.groups()
+
+        next_line, next_match, multi_line_message = peak_multi_line_message(log, pattern)
+
+        # Mypy can't determine length of regex expressions
+        partial_log_record = PartialLogRecord(*groups, message + multi_line_message)  # type: ignore
+
+        yield partial_log_record
+
+        line, match = next_line, next_match
+
+
+class LogsRecorder:
+    def __init__(
+        self,
+        install_dir: Path | str,
+        workflow: str,
+        job_id: int,
+        job_run_id: int,
+        sql_backend: SqlBackend,
+        schema: str,
+        *,
+        minimum_log_level: int = logging.WARNING,
+    ):
+        """
+        Initializes a LogProcessor instance.
+
+        Args:
+            install_dir (Path | str): The installation folder on WorkspaceFS
+            workflow (str): The workflow name.
+            job_id (int): The job id of the job to store the log records for.
+            job_run_id (int): The job run id of the job to store the log records for.
+            sql_backend (SqlBackend): The SQL Execution Backend abstraction (either REST API or Spark)
+            schema (str): The schema name for the logs persistence.
+            minimum_log_level (int) : The minimum log level to record, all records with a lower log level are excluded.
+        """
+        self._catalog = "hive_metastore"
+        self._table = "logs"
+
+        self._workflow = workflow
+        self._job_id = job_id
+        self._job_run_id = job_run_id
+        self._sql_backend = sql_backend
+        self._schema = schema
+        self._minimum_log_level = minimum_log_level
+
+        self.log_path = Path(install_dir) / "logs" / self._workflow / f"run-{self._job_run_id}"
+
+    @property
+    def full_name(self) -> str:
+        return f"{self._catalog}.{self._schema}.{self._table}"
+
+    def record(self, task_name: str, log: TextIO, log_creation_timestamp: datetime.datetime) -> list[LogRecord]:
+        """Record the logs of a given task.
+
+        Args:
+            task_name (str): The name of the task
+            log (TextIO): The task log
+            log_creation_timestamp (datetime.datetime): The log creation timestamp.
+        """
+        log_records = [
```

Comment:
if you're running this as the last task, then `Path(install_dir) / "logs" / self._workflow / f"run-{self._job_run_id}"` holds logs from all tasks. read them all, that's where you can parse the task names - remember - files won't always have `.log` extension, because they're rotated.